commands/root.go

```diff
@@ -15,6 +15,12 @@ var GitCommit string
 var GitLastTag string
 var GitExactTag string
 
+type GlobalOptions struct {
+	nonInteractive bool
+}
+
+var rootOptions GlobalOptions
```

Comment:
I get why you did  that but we really should avoid global variables like that. At the beginning of git-bug each flags were done as global variables like that (following the cobra standard) but it ended up being a major pain. Following that came this new "functional" structure without any globals.

Now to get out of this, there still might be a way to define the flag at the root level (say, with `cmd.PersistentFlags().Bool("non-interactive", ....)`, but it looks to me that we could just avoid that entirely and define the flag only on the commands where it makes sense (that is, those that have prompts).

What do you think?

Reply:
I have the same feeling about the current solution. Good that we are on the same wavelength.
I will remove the global option and instead add a dedicated non-interactive option on each of the interactive commands.
Shouldn't be to much work.